### DISCUSSION TITLE
Traceability

### DISCUSSION THREAD

**Discussion by fernandocalsa** [Budibase/budibase#6313] [Ideas]
Hi! I'm moving this from the [issue](https://github.com/Budibase/budibase/issues/6219) to here, as I think it makes more sense to discuss it here. Please let me know if I'm wrong.

Hi! I'm creating my first apps with Budibase. For the project I'm working on right now, we require to have traceability of all the actions a user has made in the app.

I've been looking around and I couldn't find anything like that, if there is something similar please let me know.

I'm thinking in a configuration per app or organisation, where the administrator can turn it on/off. If it is enabled, Budibase will create a table in Budibase database with these columns:
- id
- userId
- queryId
- bindings
- timestamp

With this schema, every time a user executes a query, Budibase will create a new row in this table with the data.

I think it can be a great feature to add into Budibase which can attract new customers with this requirement.

**Comment by melohagan** [comment]
This is a great idea!

Q4 of the roadmap has plans for expanding upon the current RBAC system, so maybe this could be a part of that: https://github.com/Budibase/budibase/projects/10
